flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/MultipleInputStreamTaskChainedSourcesCheckpointingTest.java

```diff
@@ -380,4 +446,57 @@ private void processSingleStepUntil(
         }
         assertTrue(condition.get());
     }
+
+    static class LifeCycleMonitorMultipleInputOperator
+            extends MultipleInputStreamTaskTest.MapToStringMultipleInputOperator {
+
+        private final StreamTaskTest.LifeCycleMonitor lifeCycleMonitor =
+                new StreamTaskTest.LifeCycleMonitor();
```

Comment:
nit: Let's move the `LifeCycleMonitor` to a top level file.

Reply:
BTW, I like the design of the class :+1:

Reply:
Very thanks~ I extracted it to a separate class~